server/api/google_forms_handler.post.ts

```diff
@@ -0,0 +1,122 @@
+import { createKysely } from '@vercel/postgres-kysely'
+import { createError } from 'h3'
+import { sql } from 'kysely'
+import type { Database } from '~/types/db/Database'
+import type ParticipantTable from '~/types/db/Participants'
+import type { ParticipantRow } from '~/types/db/Participants'
+import type { RobotRow } from '~/types/db/Robots'
+import type RobotTable from '~/types/db/Robots'
+import type { TeamRow } from '~/types/db/Teams'
+import type TeamTable from '~/types/db/Teams'
+import type TeamsParticipantsTable from '~/types/db/TeamsParticipants'
+import { type TeamsParticipantsRow } from '~/types/db/TeamsParticipants'
+import { isFormsResponse, type FormsResponse, type Participant, type Robot } from '~/types/Forms_response'
+
+function insertUpdateTeam(team_name: string) {
+  return sql<TeamTable>`robocomp.fn_insert_update_team(${sql.lit(team_name)})`
+}
+function insertUpdateParticipant(participant: Participant) {
+  return sql<ParticipantTable>`robocomp.fn_insert_update_participant(${sql.lit(participant.first_name)},${sql.lit(participant.last_name)},${sql.lit(participant.email)},${sql.lit(participant.phone)},${sql.lit(participant.street_address)},${sql.lit(participant.admin_level_2)},${sql.lit(participant.postal_code)},${sql.lit(participant.country)})`
+}
+//the robot numer must be unique
+function insertUpdateRobot(robot: Robot, team_id: number) {
+  return sql<RobotTable>`robocomp.fn_insert_update_robot(robocomp.fn_get_next_robot_no(),${sql.lit(robot.name)},2024::smallint,${sql.lit(team_id)},${sql.lit(robot.competition)})`
+}
+
+function connectTeamParticipant(participant_id: number, team_id: number, role: string) {
+  return sql<TeamsParticipantsTable>`robocomp.fn_connect_participant_and_team(${sql.lit(participant_id)},${sql.lit(team_id)},${sql.lit(role)})`
+}
+
+function getNextRobotNo() {
+  return sql`robocomp.fn_get_next_robot_no()`
+}
+
+export default defineEventHandler(async (event) => {
+  const rawBody = await readBody(event)
+  if (!isFormsResponse(rawBody)) {
+    throw 'Invalid form data'
+  }
+  const body = rawBody as FormsResponse
+
+  const db = createKysely<Database>()
+
+  // there are no clear ways to check if google forms was edited, so this will have to be done by checking if the participant email was already used.
+  try {
+    const participant_team_id = await db
+      .selectFrom('robocomp.participants')
+      .select('id')
+      .where('email', '=', body.participants[0].email)
+      .execute()
+    if (participant_team_id.length !== 0) {
+      throw 'Edits are not yet supported'
+    }
+    // Create a team
+    const team = await sql<TeamRow[]>`SELECT * FROM ${insertUpdateTeam(body.team_name)}`.execute(db)
```

Comment:
change from `sql<TeamRow[]>` to `sql<{ fn_insert_update_team: number }[]>`

Reply:
Still doesn't work:
Property 'fn_insert_update_team' does not exist on type '{ fn_insert_update_team: number; }[]'.
on line:
const team_id = team.rows[0].fn_insert_update_team

Reply:
oh, right, try without square brackets (`sql<{ fn_insert_update_team: number }>`)

`const team` is already an array, I forgot about what `execute` function returns